src/context/ThemeContext.js

```diff
@@ -0,0 +1,18 @@
+import React,{createContext} from 'react'
+import useLocalStorage from '../hooks/useLocalStorage'
+
+const ThemeContext = createContext();
+// Provider is the component that will be used to wrap the app
+//this uses the useLocalStorage hook to set the theme and set the value in local storage
+//the value is set to light by default
+export const ThemeProvider = ({children}) => {
+    const [theme, setTheme] = useLocalStorage('theme', 'light')
+    const toggleTheme = () => {
+        setTheme(theme === 'light' ? 'dark' : 'light')
```

Comment:
it's better to take a boolean flag isDark rather than comparing theme everytime

Reply:
Will Improvise